.github/workflows/cloud-tpu-ci-nightly.yml

```diff
@@ -0,0 +1,55 @@
+name: Cloud TPU nightly
+
+on:
+  schedule:
+    - cron: "0 14 * * *"  # daily at 7am PST
+  workflow_dispatch:  # allows triggering the workflow run manually
+
```

Comment:
Do you want to set permissions for the GitHub token?

Reply:
They're already set to read-only via the project settings. Should I also set it here?

Reply:
That's sufficient, but it's best practice to also include them here, in case someone modifies the project settings later on.

Reply:
Got it, done. I put `contents: read`, that seems to be sufficient (https://github.com/skye/jax/actions/runs/3365397735/jobs/5580821741). It's unclear to me if it even needs that, or if it gets the current checkout "for free".

Reply:
I haven't heard of that best practice before. I guess it makes some sense, but seems a bit verbose. The people changing the project settings should be at least repository admins, and if they change the default settings it's presumably because that's the goal they're trying to achieve. In IREE, I've actually set the default to read-only at the organization level and that appears to make it impossible to set to write at the repo level (which is actually odd; I thought that was only supposed to set the default). I guess this is more explicit in that it documents that this workflow only needs read access to the repository contents and not to releases or whatever. I don't know if I actually have a point...

Reply:
config-as-code is nice to have during code / audit reviews and incidence response, especially that it's part of the commit history... unlike the settings.

Reply:
Yeah, fair